### ISSUE TITLE
separate https from http activity for squid-deb-proxy-client?

### ISSUE THREAD

**Issue by kingdonb** [phusion/passenger-docker#10]
I have a fork of [phusion/baseimage-docker](github.com/phusion/baseimage-docker) where I've added in support for my own local squid-deb-proxy server, which speeds up rebuild of images immeasurably.

The phusion passenger mirror site is HTTPS-only and 302's the client if it tries to access any URL via un-crypted channels.  I was trying to think of a way to deal with this and the best solution I can come up with is to do most of the work where proxy can be used (archive.ubuntu.com, ppa.launchpad.net) upfront, then add the HTTPS mirrors later and install the passenger debs only once the proxy has been disabled.  Still this is not the best solution.

If this is too far out of scope feel free to close this issue with no action.  I can't think of any good solution; maybe there is a way to instruct Apt to only use the proxy for HTTP plain text, it seems like The Internet Says that you have to explicitly enable the proxy for https, which I am not doing.  http://chandrusoft.wordpress.com/2012/09/30/disable-proxy-when-using-apt-get/

I don't really want to MITM my connection with oss-binaries.phusionpassenger.com
